static/espeasy.js

```diff
@@ -1,11 +1,41 @@
+  "ControllerEnable", "DateTime", "Debug", "DeepSleep", "DNS", "DST", "EraseSDKwifi", "ExecuteRules", "Gateway", "I2Cscanner",
+  "IP", "Let", "Load", "LogEntry", "LogPortStatus", "LoopTimerSet", "LoopTimerSet_ms", "MemInfo", "MemInfoDetail", "Name", "Password", "Publish",
+  "Reboot", "Reset", "ResetFlashWriteCounter", "Save", "SendTo", "SendToHTTP", "SendToUDP", "Settings", "Subnet", "Subscribe", "TaskClear", "TaskClearAll",
+  "TaskDisable", "TaskEnable", "TaskRun", "TaskValueSet", "TaskValueSetAndRun", "TimerPause", "TimerResume", "TimerSet", "TimerSet_ms", "TimeZone",
+  "UdpPort", "UdpTest", "Unit", "UseNTP", "WdConfig", "WdRead", "WifiAPkey", "WifiAllowAP", "WifiAPMode", "WifiConnect", "WifiDisconnect", "WifiKey",
+  "WifiKey", "WifiScan", "WifiSSID", "WifiSSID2", "WifiSTAMode",
+  "Event", "AsyncEvent", "/control?cmd",
+  "GPIO", "GPIOToggle", "LongPulse", "LongPulse_mS", "Monitor", "Pulse", "PWM", "Servo", "Status", "MCPGPIO", "MCPGPIOToggle", "MCPLongPulse",
+  "MCPLongPulse_ms", "MCPPulse", "Status,MCP", "Monitor,MCP", "UnMonitor,MCP", "MonitorRange,MCP", "UnMonitorRange,MCP", "MCPGPIORange", "MCPGPIOPattern",
+  "MCPMode", "MCPModeRange", "PCFGPIO", "PCFGPIOToggle", "PCFLongPulse", "PCFLongPulse_ms", "PCFPulse", "Status,PCF", "Monitor,PCF", "UnMonitor,PCF",
+  "MonitorRange,PCF", "UnMonitorRange,PCF", "PCFGPIORange", "PCFMode", "Tone", "RTTTL", "UnMonitor",];
+var commonString2 = ["Clock#Time", "Login#Failed", "MQTT#Connected", "MQTT#Disconnected", "MQTTimport#Connected", "MQTTimport#Disconnected", "Rules#Timer", "System#Boot",
+  "System#BootMode", "System#Sleep", "System#Wake", "TaskExit#", "TaskInit#", "Time#Initialized", "Time#Set", "WiFi#APmodeDisabled", "WiFi#APmodeEnabled",
+  "WiFi#ChangedAccesspoint", "WiFi#ChangedWiFichannel", "WiFi#Connected"];
+var commonString3 = ["ResetPulseCounter", "SetPulseCounterTotal", "LogPulseStatistic", "LCDCmd", "LCD", "OledFramedCmd", "OledFramedCmd,Display", "OledFramedCmd,Frame",
+  "MotorShieldCmd,DCMotor", "MotorShieldCmd,Stepper", "Sensair_SetRelay", "PMSX003,Wake", "PMSX003,Sleep", "PMSX003,Reset", "Play", "Vol", "Eq", "Mode", "Repeat",
+  "HLWCalibrate", "HLWReset", "WemosMotorShieldCMD", "LolinMotorShieldCMD", "HeatPumpir", "MitsubishiHP,temperature", "MitsubishiHP,power", "MitsubishiHP,mode",
+  "MitsubishiHP,fan", "MitsubishiHP,vane", "MitsubishiHP,widevane", "Culreader_write", "TFTcmd", "TFT&", "Touch,Rot", "WakeOnLan", "Max1704xclearalert"];
+var commonTag = ["On", "Do", "Endon"];
+var commonNumber = ["toBin", "toHex", "Constrain", "XOR", "AND:", "OR:", "Ord", "bitRead", "bitSet", "bitClear", "bitWrite", "urlencode"];
+var commonMath = ["Log", "Ln", "Abs", "Exp", "Sqrt", "Sq", "Round", "Sin", "Cos", "Tan", "aSin", "aCos", "aTan", "Sind_d", "Cos_d", "Tan_d", "aSin_d", "aCos_d", "sTan_d"];
+var commonWarning = ["delay", "Delay", "Yehudit"];
+var AnythingElse = [];
+
```

Comment:
As a general remark on these lists: It will be helpful to group the keywords, f.e. by topic or plugin, and put them on a separate line, with the closing `];` also on a separate line, to make merging in extra/new (plugin-specific) keywords easier. Then it is more attractive for developers to add new keywords to the lists, as they will then also be highlighted in the editor. 😉

Reply:
Good, idea... therefore i need to figure out which keyword of the existing list belongs to which plugin

but you mean it like this?:

```
var commonString2 = [
//plugin123
"Clock#Time", "Login#Failed", "MQTT#Connected", "MQTT#Disconnected", "MQTTimport#Connected", 
//plugin234
"MQTTimport#Disconnected", "Rules#Timer", "System#Boot",
 ];
```